packages/create-subscription/src/__tests__/createSubscription-test.internal.js

```diff
@@ -0,0 +1,423 @@
+/**
+ * Copyright (c) 2013-present, Facebook, Inc.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @emails react-core
+ */
+
+'use strict';
+
+let createSubscription;
+let ReactFeatureFlags;
+let React;
+let ReactNoop;
+
+describe('createSubscription', () => {
+  beforeEach(() => {
+    jest.resetModules();
+    createSubscription = require('create-subscription').createSubscription;
+    ReactFeatureFlags = require('shared/ReactFeatureFlags');
+    ReactFeatureFlags.debugRenderPhaseSideEffectsForStrictMode = false;
+    React = require('react');
+    ReactNoop = require('react-noop-renderer');
+  });
+
+  // Mimics a partial interface of RxJS `BehaviorSubject`
+  function createFauxBehaviorSubject(initialValue) {
+    let currentValue = initialValue;
+    let subscribedCallbacks = [];
+    return {
+      getValue: () => currentValue,
+      subscribe: callback => {
+        subscribedCallbacks.push(callback);
+        return {
+          unsubscribe: () => {
+            subscribedCallbacks.splice(
+              subscribedCallbacks.indexOf(callback),
+              1,
+            );
+          },
+        };
+      },
+      update: value => {
+        currentValue = value;
+        subscribedCallbacks.forEach(subscribedCallback =>
+          subscribedCallback(value),
+        );
+      },
+    };
+  }
+
+  // Mimics a partial interface of RxJS `ReplaySubject`
+  function createFauxReplaySubject(initialValue) {
+    const observable = createFauxBehaviorSubject(initialValue);
+    const {getValue, subscribe} = observable;
+    observable.getValue = undefined;
+    observable.subscribe = callback => {
+      callback(getValue());
+      return subscribe(callback);
+    };
+    return observable;
+  }
+
+  it('supports basic subscription pattern', () => {
+    const Subscription = createSubscription({
+      getValue: source => source.getValue(),
+      subscribe: (source, callback) => source.subscribe(callback),
+      unsubscribe: (source, subscription) => subscription.unsubscribe(),
+    });
+
+    const observable = createFauxBehaviorSubject();
+    ReactNoop.render(
+      <Subscription source={observable}>
+        {(value = 'default') => {
+          ReactNoop.yield(value);
+          return null;
+        }}
+      </Subscription>,
+    );
+
+    // Updates while subscribed should re-render the child component
+    expect(ReactNoop.flush()).toEqual(['default']);
+    observable.update(123);
+    expect(ReactNoop.flush()).toEqual([123]);
+    observable.update('abc');
+    expect(ReactNoop.flush()).toEqual(['abc']);
```

Comment:
ReactNoop makes this kind of test so much easier :)

Reply:
Yes agreed!